src/Catrobat/Controller/Web/NotificationsController.php

```diff
@@ -34,8 +36,6 @@ public function NotificationsAction(CatroNotificationRepository $notification_re
 
     $catro_user_notifications = $notification_repository->findBy(['user' => $user], ['id' => 'DESC']);
```

Comment:
you might wanna get rid of the whole fetch notification logic here and only work with the API GET request to fetch notifications.

Load the page without notifications.
In JS Trigger the load of the first 20 notifications of the category that is visible.
When switching to different category load the first 20 notifications. Or you might wanna load them already once everything for the first category finished. (In order to utilize the whole computation power of the client)